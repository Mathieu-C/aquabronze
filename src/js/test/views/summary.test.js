import cartService from "../../services/cart";
import summaryView from "../../views/summary";
import cartItems from "../fixtures/cartItems";

beforeEach(() => {
	cartService.clear();
});

describe("summary view", () => {
	describe("render", () => {
		it("renders the summary", () => {
			const domTarget = document.querySelector("#js-cart-summary");
			summaryView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});

		it("reflects cart updates correctly in the summary", () => {
			cartService.populate(cartItems);
			const domTarget = document.querySelector("#js-cart-summary");
			summaryView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});

		it("renders the cart total", () => {
			const domTarget = document.querySelector("#js-cart-total");
			summaryView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});

		it("reflects cart updates correctly in the cart total", () => {
			cartService.populate(cartItems);
			const domTarget = document.querySelector("#js-cart-total");
			summaryView.render();
			expect(domTarget.innerHTML).toMatchSnapshot();
		});
	});
});
