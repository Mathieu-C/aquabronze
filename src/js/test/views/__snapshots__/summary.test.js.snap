// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary view render reflects cart updates correctly in the cart total 1`] = `
"
      <span>Total cost</span>
      <span class=\\"summary__total-net\\">32.5 €</span>
    "
`;

exports[`summary view render reflects cart updates correctly in the summary 1`] = `
"
      <span>4 Items</span>
      <span class=\\"summary__count-gross\\">37.5 €</span>
    "
`;

exports[`summary view render renders the cart total 1`] = `
"
      <span>Total cost</span>
      <span class=\\"summary__total-net\\">0 €</span>
    "
`;

exports[`summary view render renders the summary 1`] = `
"
      <span>0 Items</span>
      <span class=\\"summary__count-gross\\">0 €</span>
    "
`;
